### DISCUSSION TITLE
I can't get it to work, please assist me

### DISCUSSION THREAD

**Discussion by CyberSlither** [CogentRedTester/mpv-sub-select#27] [Q&A] [answered]
Since the start, i've had only 3 folders in the bootstraper folder where mpv.exe is and the folders are "doc", "installer" and "mpv"
I had to make a portable_config myself in which I have the files "input.conf" and "mpv.conf" which I also made by myself
I don't have any files nor folders in the path C:\Users\(username)\AppData\Roaming\mpv and its empty
I tried doing what I read but can't seem to get it to work probably because I can't create the folders or paste the files in the right places but could be something else as well
My mpv.conf file has the following code

`volume=100
volume-max=200
save-position-on-quit
fullscreen=yes
alang=jpn,eng
slang=eng
sub=auto`

I want to set it to always select the subtitles with the word "dialogue" in them or to ignore the ones with the words "signs" and "songs" in them
I've provided with all the information I could think of and eagerly for your assistance
Please guide me for where to create folders and paste which files where as well
You have my thanks

**Comment by CogentRedTester** [comment]
This is the file structure, the top-level `mpv/` directory is the equivalent of either `C:\Users(username)\AppData\Roaming\mpv` or the `portable_config` directory:

```tree
mpv/
├── script-opts/
│   ├── sub_select.conf
│   └── sub-select.json
├── scripts/
│   └── sub-select.lua
├── input.conf
└── mpv.conf

```

Try using the default `sub-select.json` file, that may do what you want already.

**Reply by CyberSlither** [reply]
Hi, thanks for a quick reply.
I tried placing the files in their original states in the order described above but it still didn't work.
Am I supposed to make any changes to the sub_select.conf ?
Once again, thanks for your help.
The signs and songs subtitles from the anime I was watching had the name "Signs & Songs" with the inverted commas inclusive, while the default sub-select.json file has Sign in the blacklist. Could that be he potential reason?
BTW, these are the the names of the subtitle track I currently want to to watch and it isn't working on this as well
![Screenshot 2024-02-01 001757](https://github.com/CogentRedTester/mpv-sub-select/assets/144510745/90ec2270-11c5-453f-9d8c-518f79ec931b)

**Reply by CogentRedTester** [reply]
Sorry for the late reply. If you send me the contents of your `sub_select.conf` and `sub-select.json` I may be able to help.

**Reply by CyberSlither** [reply]
*I'm just happy there's a reply,
These are the contents of my `sub_select.conf`*

```
#######################################################
## Default configuration file for mpv-sub-select     ##
## https://github.com/CogentRedTester/mpv-sub-select ##
#######################################################

# forcibly enable the script regardless of the sid option
force_enable=no

# selects subtitles synchronously during the preloaded hook, which has better
# compatability with other scripts and options
# this requires that the script predict what the default audio track will be,
# so theoretically this can be wrong on some rare occasions
# disabling this will switch the subtitle track after playback starts
preload=yes

# experimental audio track selection based on the preference json file.
# this overrides force_prection and detect_incorrect_predictions.
select_audio=no

# remove any potential prediction failures by forcibly selecting whichever
# audio track was predicted
force_prediction=no

# detect when a prediction is wrong and re-check the subtitles
# this is automatically disabled if `force_prediction` is enabled
detect_incorrect_predictions=yes

#observe audio switches and reselect the subtitles when alang changes
observe_audio_switches=no

# Only select forced subtitles if they are explicitly stated in slang.
# By default, when searching for subtitle tracks with a specific language,
# forced subtitles will be included in the search results and treated the same as other tracks.
# This means that there's no way to write a rule that specifically excludes
# forced subtitle tracks. By enabling this forced subtitles will never be chosen unless a rule
# explicitly includes "forced" in `slang`.
explicit_forced_subs=no

# the folder that contains the 'sub-select.json' file
config=~~/script-opts

```
*And these are the contents of my `sub-select.json`*
```
[
    {
        "alang": ["eng?", "und"],
        "slang": ["eng?", "und"],
        "condition": "sub.forced"
    },
    {
        "alang": ["jpn", "ja"],
        "slang": "eng?",
        "blacklist": [ "sign" ]
    },
    {
        "inherit": "^",
        "slang": "und"
    },
    {
        "alang": "eng?",
        "slang": [ "eng?", "und" ],
        "whitelist": [ "sign", "song" ]
    },
    {
        "alang": "eng?",
        "slang": "no"
    },
    {
        "alang": "*",
        "slang": [ "eng?", "und" ]
    },
    {
        "alang": "no",
        "slang": "no"
    }
]
```
> 
[sub-select.json](https://github.com/CogentRedTester/mpv-sub-select/files/14589963/sub-select.json)
 <
 *I've uploaded the `sub-select.json` just in case
 It doesn't let me upload the `sub-select.conf` file so i'm uploading it as `sub-select.txt`*
[sub_select.txt](https://github.com/CogentRedTester/mpv-sub-select/files/14590167/sub_select.txt)

**Reply by CogentRedTester** [reply]
Looks correct, I wonder if the script files are actually being loaded properly. Can you please upload a log file generated with: `--log-file=log.txt --msg-level=sub_select='debug'`

**Reply by CyberSlither** [reply, marked as answer]
Hi, sorry for the late reply, I finally managed to fix thanks to checking the log file which I was about to upload just now
what the issue turned out to be was that the 'script-opts' folder was named incorrectly as 'scripts-ops' and the sub-select files weren't being detected😅
The issue was entirely on my side and due to human error so I apologize for wasting your time but in the end the reason I was able to point out the problem was thanks to your help.
Thanks a lot and i'm grateful that you spent the time that you had no obligation to spend helping me 😄

**Reply by pegasusearl** [reply]
I felt like this should be written in readme.md. For those who are not familiar with how mpv script works, figuring out how to install it is challenging. Especially since the files in this repository is located in the same directory.

I'm talking about me, I can't figure out how to install this before I saw your file structure post. Luckily I found it here, yay celebration!
